Add support for ternary conditional expression in tail call optimization

kiendang opened the pull request:
This PR enables tail call optimization for ternary conditional expression in return statement.
e.g
```python
@tco
def tail_rec(x, acc=0):
    return acc if len(x) == 0 else tail_rec(x[1:], acc + x[0])
```

azazel75 commented:
@kiendang thanks  for the contribution, can you please add a test for this?.

kiendang commented:
I already did?

kiendang commented:
Sorry please don't merge yet. It's not perfect. I'll update + add more tests soon.

kiendang commented:
@azazel75 I've finished. Please review when you have time.

```python
@tco
def tail_rec(x, acc=0):
    return acc if len(x) == 0 else tail_rec(x[1:], acc + x[0])
```
will now expand to
```python
return (acc if len(x) == 0 else (TCOType.CALL, tail_rec, [x[1:], acc + x[0]], {}))
```
I also added support for non-return ternary expression in tail position and added more tests (basically all the existing tests for tco rewritten using ternary expression).
